File: 2022-2023/sourcecode/lab/lab7/GalleryAppJava/app/src/main/java/pl/udu/uwr/pum/galleryappjava/adapter/GalleryAdapter.java
package pl.udu.uwr.pum.galleryappjava.adapter;

import android.net.Uri;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import pl.udu.uwr.pum.galleryappjava.databinding.ItemViewBinding;
import pl.udu.uwr.pum.galleryappjava.model.PictureModel;

public class GalleryAdapter extends RecyclerView.Adapter<GalleryAdapter.ViewHolder> {

    private final ArrayList<PictureModel> pictures;

    public GalleryAdapter(ArrayList<PictureModel> pictures){
        this.pictures = pictures;
    }
    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(ItemViewBinding.inflate(
                LayoutInflater.from(parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        PictureModel item = pictures.get(position);
        holder.bind(item);
    }

    @Override
    public int getItemCount() {
        return pictures.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final ItemViewBinding itemBinding;
        public ViewHolder(@NonNull ItemViewBinding itemBinding) {
            super(itemBinding.getRoot());
            this.itemBinding = itemBinding;
        }

        public void bind(PictureModel item){
            itemBinding.textViewTitle.setText(item.getTitle());
            itemBinding.rcImageView.setImageURI(Uri.parse(item.getImage()));
        }
    }
}

File: 2022-2023/sourcecode/lab/lab7/GalleryApp/app/src/main/java/pl/udu/uwr/pum/galleryapp/adapter/GalleryAdapter.kt
package pl.udu.uwr.pum.galleryapp.adapter

import android.net.Uri
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import pl.udu.uwr.pum.galleryapp.databinding.ItemViewBinding
import pl.udu.uwr.pum.galleryapp.model.PictureModel

class GalleryAdapter(private val pictures: List<PictureModel>) : RecyclerView.Adapter<GalleryAdapter.ViewHolder>() {

    class ViewHolder(private val itemBinding: ItemViewBinding) : RecyclerView.ViewHolder(itemBinding.root) {
        fun bind (item: PictureModel){
            itemBinding.textViewTitle.text = item.title
            itemBinding.rcImageView.setImageURI(Uri.parse(item.image))
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            ItemViewBinding.inflate(
                LayoutInflater.from(parent.context), parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = pictures[position]
        holder.bind(item)
    }

    override fun getItemCount(): Int = pictures.size
}